main.py

```diff
@@ -47,35 +53,86 @@ def start_server(self, socket=os_socket):
+                                new_peers = message.get("nodes", [])
+                                all_peers.update(new_peers)
+                                all_peers.remove(self.ip)
+                                self.peer_hosts = list(all_peers)
+                            else:
+                                print(f"Received by {data}")
+                                logger.debug(f"Received by {message['sender']}: {message}")
+                                self.message_queue.put(data)
+
+                                if not self.message_queue.empty():
+                                    msg = self.message_queue.get()
 
-                                ack = json.dumps({"type": "ack", "message": f"Received {message['message']} from {message['sender']}", "sender": self.nickname}).encode()
-                                conn.sendall(ack)
-                                logger.debug(f"{self.nickname} sent ack {ack}")
+                                    ack = json.dumps({"type": "ack", "message": f"Received {message['message']} from {message['sender']}", "sender": self.nickname}).encode()
+                                    conn.sendall(ack)
+                                    logger.debug(f"{self.nickname} sent ack {ack}")
         except Exception as exc:
             logger.exception(exc)
             raise exc
 
+    def request_peers(self, peer_port, socket=os_socket):
+
+        all_peers = set(self.peer_hosts)
+        try:
+            with socket(AF_INET, SOCK_STREAM) as s:
+                s.connect(("startup_server", peer_port))
+                s.sendall(json.dumps({"type": "GET_NODES", "node_ip": self.ip}).encode())
+                data = s.recv(1024)
+
+                response = json.loads(data)
+                peers = response.get("nodes", [])
+
+                all_peers.update(peers)
+
+        except Exception as exc:
+            logger.error(f"Failed to connect to startup server: {exc}")
+
+        self.peer_hosts = list(all_peers)
+        print(f"Connected to {self.peer_hosts}")
+
+    def send_peers(self, peer_port, socket=os_socket):
+        try:
+            for peer_host in self.peer_hosts:
+                with socket(AF_INET, SOCK_STREAM) as s:
+                    s.connect((peer_host, peer_port))
+                    s.sendall(json.dumps({"type": "NEW_NODES", "nodes": self.peer_hosts}).encode())
+
+        except Exception as exc:
+            logger.error(f"Failed to send new peers: {exc}")
+
+
 # Only run this code if the file was executed from command line
 if __name__ == '__main__':
 
     if '--help' in sys.argv:
-        print(f"Usage: {sys.argv[0]} [PEER_1] [PEER_2] ...")
+        print("Start the startup server:")
+        print(f"Usage: {sys.argv[0]} startup")
         exit(-1)
 
-    peer_hosts = sys.argv[1:] # svm-11-3.cs.helsinki.fi
-    nickname = input("Set nickname: ")
+    if len(sys.argv) > 1 and sys.argv[1] == "startup":
+        startup = 1
+        peer_hosts = None
```

Comment:
Why not initialize this with empty list, then you don't need to handle it special way in Node constructor
```suggestion
        peer_hosts = []
```